x-pack/plugin/migrate/src/main/java/org/elasticsearch/xpack/migrate/action/CancelReindexDataStreamAction.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.migrate.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class CancelReindexDataStreamAction extends ActionType<CancelReindexDataStreamAction.Response> {
+
+    public static final CancelReindexDataStreamAction INSTANCE = new CancelReindexDataStreamAction();
+    public static final String NAME = "indices:admin/data_stream/reindex_cancel";
+
+    public CancelReindexDataStreamAction() {
+        super(NAME);
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        public Response() {
+            super();
+        }
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {}
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field("acknowledged", true);
```

Comment:
Could this action just use `AcknowledgeResponse` directly?

Reply:
Oh yeah, it could! In an early iteration I was returning the latest status information so the response had fields in it. But then I decided that was unnecessary and forgot to remove the response class.